src/Common/ShellCommandSettings.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+
+namespace DB
+{
+
+enum class ExternalCommandStderrReaction
+{
+    NONE, /// Do nothing.
+    LOG, /// Try to log all outputs of stderr from the external command.
+    THROW /// Throw exception when the external command outputs something to its stderr.
+};
+
+ExternalCommandStderrReaction parseExternalCommandStderrReaction(const std::string & config);
+
+enum class ExternalCommandErrorExitReaction
+{
+    NONE, /// Do nothing.
+    LOG_FIRST, /// Try to log first 1_KiB outputs of stderr from the external command.
+    LOG_LAST /// Same as above, but log last 1_KiB outputs.
+};
```

Comment:
this is very strange for couple of reasons:
1. why to split it like this? why not have one with options `NONE, LOG, LOG_FIRST, LOG_LAST, THROW`
2. `!= ExternalCommandErrorExitReaction::NONE` is used to define reaction to exit code - it should not be dependent on stderr logging at all, it should be an independent setting I think. See below for note regarding exit code.

Reply:
Because `LOG_FIRST` and `LOG_LAST` are reactions to exit code instead of stderr.

Reply:
If we specify `ExternalCommandStderrReaction::LOG`, it will log every content of stderr no matter what exit code is. OTOH, `ExternalCommandErrorExitReaction::LOG_FIRST` will only log the first 4_KiB of stderr when seeing non-zero exit code. If we merge `LOG_FIRST`, `LOG_LAST` into `ExternalCommandStderrReaction`, it will be inconsistent.

Maybe we only need one setting to turn on stderr logging (because why not log everything?), and one setting to check exit code.

Reply:
according to code ExternalCommandStderrReaction::LOG will LOG_WARNING all stderr output - it doesn't depend on ExternalCommandErrorExitReaction. And only later we LOG_ERROR according to ExternalCommandErrorExitReaction. Kind of messy I think. probably having all stderr related stuff in one place will be better -
ExternalCommandStderrReaction::
NONE - do nothing - no logging, no throw
LOG - LOG_WARNING everything
LOG_FIRST, LOG_LAST - LOG_WARNING first, last
THROW - throw immediately with whatever first

and left ExternalCommandErrorExitReaction for explicit reaction for exit code != 0